### PR TITLE
Fix integration tests

### LINKED ISSUE #1280: Integration tests failing
## What went wrong? 😲
Integration tests are failing

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1280

# 👩🏻‍💻 What does this PR do? 
Fixes integration tests

# 🧪 How has/should this change been tested? 
- Set up 4d for integration test
- Run integration test in remote server

### CODE REVIEW

**Comment by andreievg** [comment]
I had a quick test with latest develop (mSupply), there was an error in item, due to another field being added to item in mSupply. This would be good to fix and add to documentation of possible things that go wrong and how to fix, this is how i debugged it:

Error was like this:

```
test_central_sync_record_unit_and_item_init
test_central_sync_record_unit_and_item_step1
thread 'sync::test::integration::central::test::tests::integration_sync_central_unit_and_item' panicked at 'assertion failed: `(left == right)`
  left: `ItemRow { id: "35c7b792-66c2-4204-8f7b-2b0522aab330", name: "5122f0bc-114f-4ed6-864e-129043a1e6cb", code: "fe9dc00d-5fdf-45d6-a289-db2dac11297c", unit_id: None, default_pack_size: 1, type: NonStock, legacy_record: "{\"ABC_category\":\"\",\"ID\":\"35c7b792-66c2-4204-8f7b-2b0522aab330\",\"Message\":\"\",\"VEN_category\":\"\",\"account_income_ID\":\"EF34ADD07C014AB8914E30CA2E3FEA8D\",\"account_purchases_ID\":\"330ACC81721C4126BD5DD6769466C5C4\",\"account_stock_ID\":\"52923505A91447B9923BA34A4F332014\",\"atc_category\":\"\",\"barcode_spare\":\"\",\"builds_only\":false,\"buy_price\":0,\"catalogue_code\":\"\",\"category2_ID\":\"\",\"category3_ID\":\"\",\"category_ID\":\"\",\"code\":\"fe9dc00d-5fdf-45d6-a289-db2dac11297c\",\"critical_stock\":false,\"cross_ref_item_ID\":\"\",\"custom_data\":null,\"ddd factor\":0,\"ddd_value\":\"\",\"default_pack_size\":1,\"department_ID\":\"\",\"description\":\"\",\"dose_picture\":\"\",\"doses\":0,\"essential_drug_list\":false,\"expiry_date_mandatory\":false,\"flags\":\"\",\"indic_price\":0,\"instructions\":\"\",\"interaction_group_ID\":\"\",\"is_sync\":false,\"is_vaccine\":false,\"item_name\":\"5122f0bc-114f-4ed6-864e-129043a1e6cb\",\"kit_data\":null,\"manufacture_method\":\"\",\"margin\":0,\"medication_purpose\":\"\",\"non_stock_name_ID\":\"\",\"normal_stock\":false,\"other_names\":\"\",\"outer_pack_size\":0,\"price_editable\":false,\"print_units_in_dis_labels\":false,\"product_specifications\":\"\",\"reference_bom_quantity\":0,\"restricted_location_type_ID\":\"\",\"sms_code\":\"\",\"sms_pack_size\":0,\"spare_hold_for_issue\":false,\"spare_hold_for_receive\":false,\"spare_ignore_for_orders\":false,\"spare_internal_analysis\":0,\"spare_non_stock\":false,\"spare_pack_to_one_on_receive\":false,\"start_of_year_date\":\"0000-00-00\",\"strength\":\"\",\"type_of\":\"non_stock\",\"unit_ID\":\"\",\"universalcodes_code\":\"\",\"universalcodes_name\":\"\",\"use_bill_of_materials\":false,\"user_field_1\":\"\",\"user_field_2\":\"\",\"user_field_3\":\"\",\"user_field_4\":false,\"user_field_5\":0,\"user_field_6\":\"\",\"user_field_7\":false,\"volume_per_outer_pack\":0,\"volume_per_pack\":0,\"warning_quantity\":0,\"weight\":0}" }`,
 right: `ItemRow { id: "35c7b792-66c2-4204-8f7b-2b0522aab330", name: "5122f0bc-114f-4ed6-864e-129043a1e6cb", code: "fe9dc00d-5fdf-45d6-a289-db2dac11297c", unit_id: None, default_pack_size: 1, type: NonStock, legacy_record: "{\"ABC_category\":\"\",\"ID\":\"35c7b792-66c2-4204-8f7b-2b0522aab330\",\"Message\":\"\",\"VEN_category\":\"\",\"account_income_ID\":\"EF34ADD07C014AB8914E30CA2E3FEA8D\",\"account_purchases_ID\":\"330ACC81721C4126BD5DD6769466C5C4\",\"account_stock_ID\":\"52923505A91447B9923BA34A4F332014\",\"atc_category\":\"\",\"barcode_spare\":\"\",\"builds_only\":false,\"buy_price\":0,\"catalogue_code\":\"\",\"category2_ID\":\"\",\"category3_ID\":\"\",\"category_ID\":\"\",\"code\":\"fe9dc00d-5fdf-45d6-a289-db2dac11297c\",\"critical_stock\":false,\"cross_ref_item_ID\":\"\",\"custom_data\":null,\"ddd factor\":0,\"ddd_value\":\"\",\"default_pack_size\":1,\"department_ID\":\"\",\"description\":\"\",\"dose_picture\":\"\",\"doses\":0,\"essential_drug_list\":false,\"expiry_date_mandatory\":false,\"flags\":\"\",\"indic_price\":0,\"instructions\":\"\",\"interaction_group_ID\":\"\",\"is_sync\":false,\"is_vaccine\":false,\"item_name\":\"5122f0bc-114f-4ed6-864e-129043a1e6cb\",\"kit_data\":null,\"manufacture_method\":\"\",\"margin\":0,\"medication_purpose\":\"\",\"non_stock_name_ID\":\"\",\"normal_stock\":false,\"other_names\":\"\",\"outer_pack_size\":0,\"price_editable\":false,\"print_units_in_dis_labels\":false,\"reference_bom_quantity\":0,\"restricted_location_type_ID\":\"\",\"sms_code\":\"\",\"sms_pack_size\":0,\"spare_hold_for_issue\":false,\"spare_hold_for_receive\":false,\"spare_ignore_for_orders\":false,\"spare_internal_analysis\":0,\"spare_non_stock\":false,\"spare_pack_to_one_on_receive\":false,\"start_of_year_date\":\"0000-00-00\",\"strength\":\"\",\"type_of\":\"non_stock\",\"unit_ID\":\"\",\"universalcodes_code\":\"\",\"universalcodes_name\":\"\",\"use_bill_of_materials\":false,\"user_field_1\":\"\",\"user_field_2\":\"\",\"user_field_3\":\"\",\"user_field_4\":false,\"user_field_5\":0,\"user_field_6\":\"\",\"user_field_7\":false,\"volume_per_outer_pack\":0,\"volume_per_pack\":0,\"warning_quantity\":0,\"weight\":0}" }`', service/src/sync/test/mod.rs:182:29
```

I realised that something was missing from `legacy_record` field.

Checked how item is created and compared:

https://github.com/openmsupply/open-msupply/blob/af698bdf00a362cc0f7185c6c7dd7940d60d5c0e/server/service/src/sync/test/integration/central/unit_and_item.rs#L56

https://github.com/openmsupply/open-msupply/blob/af698bdf00a362cc0f7185c6c7dd7940d60d5c0e/server/service/src/sync/test/integration/central/unit_and_item.rs#L134-L136

https://github.com/openmsupply/open-msupply/blob/af698bdf00a362cc0f7185c6c7dd7940d60d5c0e/server/service/src/sync/test/test_data/item.rs#L8-L15

Used postman to sync central records using api v5 collection, and searched for "item", found it and added `    "product_specifications": ""` at the end of `ITEM_1`

Fixed

File: server/service/src/sync/test/integration/central/name_and_store_and_name_store_join.rs [hunk 1]
```diff
@@ -85,6 +85,7 @@ impl SyncRecordTester for NameAndStoreAndNameStoreJoinTester {
             name_id: name_row1.id.clone(),
             code: small_uuid(),
             site_id: new_site_properties.site_id as i32,
+            logo: None,
```
Comment (andreievg) [inline server/service/src/sync/test/integration/central/name_and_store_and_name_store_join.rs:88]: Hmm, might be good to add another store and test with something in logo

**Comment by roxy-dao** [comment]
> 89af0ae

I think that was added on after I did this PR 🙇🏻 Thanks Andrei!

**Review by andreievg — APPROVED** [review]
🙏

**Review by clemens-msupply — APPROVED** [review]
works find! thanks!

There is one warning: 
```
warning: unused import: `logging_init`
 --> server/src/main.rs:3:29
  |
3 | use server::{configuration, logging_init, start_server};
  |                             ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
```
but better leave for another PR

**Pull request merged**